packages/wrangler/src/deploy/deploy.ts

```diff
@@ -395,13 +395,13 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 
 	const minify = props.minify ?? config.minify;
 
+	const compatibilityFlags =
+		props.compatibilityFlags ?? config.compatibility_flags;
```

Comment:
nit: there are quite a few `const someField = props.someField ??/|| config.someFiled`, maybe replace with

```js
const resolvedConfig = {...props, ...config}; // x1
const {someField} = resolvedConfig; // x15
```

Reply:
I'd rather be explicit about which flags can be overridden, rather than hiding them in a general override like that.

Reply:
Also note that in `props` the properties tend to be camelCased while in `config` they can be snake_cased.